python/tank/platform/qt/login_dialog.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+try:
+    from . import QtCore, QtGui
+except ImportError:
+    try:
+        from PySide import QtCore, QtGui
+    except ImportError:
+            from PyQt import QtCore, QtGui
+
+from tank.errors import TankAuthenticationError
+
+from . import resources_rc
+from . import ui_login
+from tank.util import authentication
+
+
+class LoginDialog(QtGui.QDialog):
+    """
+    Dialog for getting user crendentials.
+    """
+    def __init__(self, title, session_renewal, parent=None, **kwargs):
+        """
+        Constructs a dialog.
+
+        :param title: Title of this dialog.
+        :param session_renewal: Configures the dialog for session renewal or user login.
+        :param parent: The Qt parent for the dialog (defaults to None)
+        :param kwargs: The following keyword arguments are recognized:
+            stay_on_top - Whether the dialog should stay on top (defaults to True)
+            hostname - The string to populate the site field with. Defaults to "".
+            http_proxy - The proxy server to use when testing authentication. Defaults to None.
+            login - The string to populate the login field with. Defaults to "".
+            pixmap - QPixmap to show in the dialog (defaults to the Shotgun logo)
+        """
+        QtGui.QDialog.__init__(self, parent)
+
+        # set the dialog to always stay on top
+        if kwargs.get("stay_on_top", True):
+            self.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint)
+
+        # setup the gui
+        self.ui = ui_login.Ui_LoginDialog()
+        self.ui.setupUi(self)
+
+        # Set the title
+        self.setWindowTitle(title)
+
+        # process kwarg overrides
+        self._http_proxy = kwargs.get("http_proxy", None)
+
+        self.ui.site.setText(kwargs.get("hostname", ""))
+        self.ui.login.setText(kwargs.get("login", ""))
+
+        # default focus
+        if self.ui.site.text():
+            if self.ui.login.text():
+                self.ui.password.setFocus()
+            else:
+                self.ui.login.setFocus()
+        else:
+            self.ui.site.setFocus()
+
+        # set the logo
+        pixmap = kwargs.get("pixmap")
```

Comment:
i think the shotgun login dialog should be totally fixed and designed and only used for this purpose. Is the idea here that would re-use the same dialog for the desktop?